### PR TITLE
Added PHP type declarations to getting started guides

### PR DESCRIPTION
**Pull request by wouterj**
Relates to #12235 

I've scanned all PHP examples in the getting started guides and updated them to use PHP typing consistently (most already used typing in some way or another).

Practical things:
* Once approved, I'll take care of merging this into 5.0 and master and open new PRs for new code blocks in those branches;
* I suggest not touching 3.4 at all, we should not update that version anymore unless there is a major bug (which is unlikely, as it's around for years already);
* Method declarations almost never change (because of the BC promise), and as we're fixing these in all active branches (except 3.4), this change will not lead to persistent merge conflicts in the future.
* I don't think we can add a DOCbot rule for this one, as we cannot introduce typing in all PHP examples everywhere (see the mentioned issue).

### CODE REVIEW

**Review by javiereguiluz — APPROVED** [review]

File: controller.rst [hunk 1]
```diff
@@ -134,7 +134,7 @@ and ``redirect()`` methods::
     use Symfony\Component\HttpFoundation\RedirectResponse;
 
     // ...
-    public function index()
+    public function index(): Response
```
Comment (derrabus) [inline controller.rst]: It seems to be common practice to add `use` statements to the code examples if a class is referenced. However, there is no such statement for `Response`. It's also missing in other examples where you've added the return type.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
Nice work. :-)

File: controller.rst (hunk 1 above)
Comment (wouterj) [reply controller.rst]: Good catch. We don't always add all use statements (only the relevant use statements, or the first occurence of that class on the page). I've rechecked the examples with `Response` return typehint and added a few use statements that I think where missing.

File: controller.rst [hunk 2]
```diff
@@ -435,8 +440,9 @@ from the session automatically as soon as you retrieve them. This feature makes
 For example, imagine you're processing a :doc:`form </forms>` submission::
 
     use Symfony\Component\HttpFoundation\Request;
+    // ...
```
Comment (derrabus) [inline controller.rst:447]:
```suggestion
    use Symfony\Component\HttpFoundation\Response;
    // ...
```

File: controller.rst [hunk 3]
```diff
@@ -614,8 +626,9 @@ The ``file()`` helper provides some arguments to configure its behavior::
 
     use Symfony\Component\HttpFoundation\File\File;
     use Symfony\Component\HttpFoundation\ResponseHeaderBag;
```
Comment (derrabus) [inline controller.rst:632]:
```suggestion
    use Symfony\Component\HttpFoundation\Response;
    use Symfony\Component\HttpFoundation\ResponseHeaderBag;
```

File: controller.rst [hunk 4]
```diff
@@ -572,7 +579,7 @@ To get the value of any :ref:`configuration parameter <configuration-parameters>
 from a controller, use the ``getParameter()`` helper method::
 
     // ...
```
Comment (derrabus) [inline controller.rst:585]:
```suggestion
    use Symfony\Component\HttpFoundation\Response;
    // ...

```

File: controller.rst [hunk 5]
```diff
@@ -402,8 +404,9 @@ To get the session, add an argument and type-hint it with
 :class:`Symfony\\Component\\HttpFoundation\\Session\\SessionInterface`::
 
     use Symfony\Component\HttpFoundation\Session\SessionInterface;
```
Comment (derrabus) [inline controller.rst:409]:
```suggestion
    use Symfony\Component\HttpFoundation\Response;
    use Symfony\Component\HttpFoundation\Session\SessionInterface;
```

File: controller.rst [hunk 6]
```diff
@@ -370,8 +371,9 @@ object. To access it in your controller, add it as an argument and
 **type-hint it with the Request class**::
 
     use Symfony\Component\HttpFoundation\Request;
+    // ...
```
Comment (derrabus) [inline controller.rst:376]:
```suggestion
    use Symfony\Component\HttpFoundation\Response;
    // ...
```

File: controller.rst [hunk 7]
```diff
@@ -201,7 +201,7 @@ If you need a service in a controller, type-hint an argument with its class
     /**
      * @Route("/lucky/number/{max}")
      */
-    public function number($max, LoggerInterface $logger)
+    public function number(int $max, LoggerInterface $logger): Response
```
Comment (derrabus) [inline controller.rst:205]: `use Symfony\Component\HttpFoundation\Response;` is missing here as well.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by wouterj** [comment]
Thanks for your detailed review @derrabus 

We often don't include the use statement in every article (most of the times, we only do so in the first example of a guide or if it's a relevant use statement). I've added some more based on your review, but skipped some others (because it was too unrelevant or already shown the use statement in code examples directly before it).

**Pull request merged**